### PR TITLE
librsvg: fix cross compilation

### PR DESCRIPTION
**Pull request by NickCao**
Fix cross compilation by explicitly passing RUST_TARGET.
Ref: https://gitlab.gnome.org/GNOME/librsvg/-/blob/main/COMPILING.md#overriding-the-rust-target-name

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] riscv64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/librsvg/default.nix [hunk 1]
```diff
@@ -78,7 +79,8 @@ stdenv.mkDerivation rec {
 
     "--enable-installed-tests"
     "--enable-always-build-tests"
-  ] ++ lib.optional stdenv.isDarwin "--disable-Bsymbolic";
+  ] ++ lib.optional stdenv.isDarwin "--disable-Bsymbolic"
+    ++ lib.optional (stdenv.buildPlatform != stdenv.hostPlatform) "RUST_TARGET=${rust.toRustTarget stdenv.hostPlatform}";
```
Comment (roberth) [inline pkgs/development/libraries/librsvg/default.nix:83]: Would it make sense for `rustc` to set this through a hook? The `zcash` package does the same.

cc rustc maintainers @madjar @cstrahan @globin @havvy

Comment (NickCao) [reply pkgs/development/libraries/librsvg/default.nix:83]: `RUST_TARGET` does not seem to be a standard envvar per https://doc.rust-lang.org/cargo/reference/environment-variables.html, most likely to be specific to these packages.

**Comment by Mindavi** [comment]
When cross-building librsvg from master it compiles, so I'm not sure what this solves. Is this PR maybe obsoleted by some other change?

**Comment by NickCao** [comment]
> When cross-building librsvg from master it compiles, so I'm not sure what this solves. Is this PR maybe obsoleted by some other change?

Rust targets only slightly differs from nixpkgs targets, thus for aarch64-linux it crosses out of the box, but that's not the case for riscv64-linux.

**Comment by Mindavi** [comment]
Totally glossed over that. I did build it for aarch64 and that also still works fine. Diff LGTM otherwise.

File: pkgs/development/libraries/librsvg/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/libraries/librsvg/default.nix:83]: https://gitlab.gnome.org/search?project_id=296&search=RUST_TARGET

**Pull request merged**